src/System.Private.Uri/src/System/UriSyntax.cs

```diff
@@ -178,26 +168,7 @@ internal bool IsAllSet(UriSyntaxFlags flags)
 
         private bool IsFullMatch(UriSyntaxFlags flags, UriSyntaxFlags expected)
         {
-            // Return true, if masking the current set of flags with 'flags' equals 'expected'.
-            // Definition 'current set of flags': 
-            // a) if updatable flags were never set: m_Flags
-            // b) if updatable flags were set: set union between all flags in m_Flags which are not updatable
-            //    (i.e. not part of c_UpdatableFlags) and all flags in m_UpdatableFlags
-
-            UriSyntaxFlags mergedFlags;
-
-            // if none of the flags in 'flags' is an updatable flag, we ignore m_UpdatableFlags
-            if (((flags & c_UpdatableFlags) == 0) || !_updatableFlagsUsed)
-            {
-                mergedFlags = _flags;
-            }
-            else
-            {
-                // mask m_Flags to only use the flags not in c_UpdatableFlags
-                mergedFlags = (_flags & (~c_UpdatableFlags)) | _updatableFlags;
-            }
-
-            return (mergedFlags & flags) == expected;
```

Comment:
Why is this all being deleted?

Reply:
This happens because `SetUpdatableFlags` is in the linker dead list, and it's the last method to set `_updatableFlags` and `_updatableFlagsUsed`, so the `if` becomes always true, so I cascased that down to the return.

I ran unit tests in the project and verified this code path is getting hit; if there's anything else I should check on please let me know. Thanks!